src/Symfony/Component/HttpFoundation/UriSigner.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+use Symfony\Component\HttpKernel\UriSigner as HttpKernelUriSigner;
+
+/**
+ * Signs URIs.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class UriSigner extends HttpKernelUriSigner
```

Comment:
we should rather duplicate the code IMHO, we can't make http-kernel a dep of http-foundation

Reply:
I remember now the only problem then would be that the `uri_signer` service need to link to the old http kernel class else it would be a bc break. So only the `Symfony\Component\HttpFoundation\UriSigner` could be wired to the none deprecated instance of UriSigner.

The good one is if we extend the `Symfony\Component\HttpKernel\UriSigner` could still be given to service already switched to the new `Symfony\Component\HttpFoundation\UriSigner` (ESI, ... could then already be switched without bc break).

Reply:
As the class is moved without any logic change, you can use `class_alias`.

As an example, see the Messenger transport classes that were moved from [`Transport`](https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineTransport.php) to [Bridges](https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/DoctrineTransport.php#L133-L135).

Also note that we should not trigger a deprecation when it's instantiated, but when it's loaded (this is implemented by the above example).

Reply:
@wouterj that is a really great solution 👍